### ISSUE TITLE
ham (hildon-application-manager) details freeze

### ISSUE THREAD

**Issue by parazyd** [maemo-leste/bugtracker#406]
HAM bugs/freezes when choosing to list details upon installing a package.

TODO: investigate and fix

The following two can probably be fixed at the same time: 

https://github.com/maemo-leste/bugtracker/issues/311
https://github.com/maemo-leste/bugtracker/issues/285

**Comment by parazyd** [comment]
https://github.com/maemo-leste/bugtracker/issues/365

**Comment by edwinsage** [comment]
I have this issue running the 20200624 image on a pinephone.  Running the App manager from the terminal gives:

    “hildon-application-manager”` terminated by signal SIGSEGV (Address boundary error)

**Comment by parazyd** [comment]
```
gdb maemo-summoner
set args /usr/bin/hildon-application-manager.launch
run
```

Probable cause per gdb:

* https://github.com/maemo-leste/hildon-application-manager/blob/master/src/details.cc#L935
* https://github.com/maemo-leste/hildon-application-manager/blob/master/src/main.cc#L498

**Comment by MerlijnWajer** [comment]
The problem seems to be that the apt worker doesn't manage to do what is asked of it, I'll debug it more later today.

**Comment by MerlijnWajer** [comment]
This is what causes it to crash: https://github.com/maemo-leste/hildon-application-manager/blob/927464b6b7aa9656442c638c86dfeadf975d1dde/src/apt-worker.cc#L3498

it seems to delete/free some memory that apt likely uses internally

**Comment by MerlijnWajer** [comment]
Fixed via ed2def0
